python/TestHarness/schedulers/Scheduler.py

```diff
@@ -0,0 +1,539 @@
+    # Note: This method ignores all errors
+    def createEdgeNodes(self, job_container_dict):
+        for tester_name, job_container in job_container_dict.iteritems():
+            tester = job_container.getTester()
+            job_dag = job_container.getDAG()
+
+            # Attempt to add this node to the DAG (prereq iteration may have already added this node)
+            job_dag.add_node_if_not_exists(job_container)
+
+            # Create the edge dependency DAG. Handle any Cyclic errors as passes (but we set the actual test as a failure).
+            for prereq in tester.getPrereqs():
+
+                # we do not care about missing prereqs
+                if prereq not in job_container_dict.keys():
+                    continue
+                else:
+                    # Attempt to add the prereq node if not already exists (main loop might have already added this node)
+                    job_dag.add_node_if_not_exists(job_container_dict[prereq])
+
+                    # Attempt to create an edge node dependency
+                    try:
+                        job_dag.add_edge(job_container_dict[prereq], job_container)
+
+                    # we do not care about cyclic errors
+                    except dag.DAGValidationError:
+                        pass
+
+        return job_container_dict
+
+    # Method to return failing testers which can only be discovered when scrutinized as a group (race conditions for example)
+    def checkGroupFailures(self, job_container_dict):
+        temp_dag = dag.DAG()
+        failed_or_skipped_testers = set([])
+
+        # Create DAG independent nodes
+        for tester_name, job_container in job_container_dict.iteritems():
+            tester = job_container.getTester()
+
+            # If this tester is not runnable, continue to the next tester
+            if not tester.getRunnable(self.options):
+                failed_or_skipped_testers.add(tester)
+                continue
+
+            # Add parent node
+            temp_dag.add_node_if_not_exists(job_container)
+
+        # Create edge nodes
+        for tester_name, job_container in job_container_dict.iteritems():
+            tester = job_container.getTester()
+
+            # Add the prereq node and edges
+            for prereq in tester.getPrereqs():
+
+                try:
+                    # This may produce a KeyError (unknown dependency)
+                    prereq_tester = job_container_dict[prereq].getTester()
+
+                    # Create the edge node (can create either a DAG validation error 'cyclic' or DAG edge not existing error 'skipped dependency')
+                    temp_dag.add_edge(job_container_dict[prereq], job_container)
+
+                # Skipped/Silent/Deleted Testers fall into this catagory, caused by 'job_container' being skipped during the first iteration above
+                except dag.DAGEdgeDepError:
+                    pass
```

Comment:
I might structure this differently, put the pass case at the bottom. That way you work your way through all the cases that you are going to actually handle followed by the cases that are left. This is fine though if you like it the way it is.